### PR TITLE
DDF-3686 Fix all UIs to respect their current context

### PR DESCRIPTION
**Pull request by brendan-hofmann**
#### What does this PR do?
Relativizes all UI calls so that DDF can be deployed behind a proxy context (e.g. https://proxy/ddf/). The ace dev server is also updated to support this. CI tooling will be added to guard against regressions.
#### Who is reviewing it? 
@SmithJosh @alexaabrd @vinamartin @djblue 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui
#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)

#### How should this be tested? (List steps with links to updated documentation)
1. Use my branch in order to configure the nginx context path (note the caveat in the readme): https://github.com/LinkMJB/nginx-reverse-proxy/pull/3
2. Install DDF with `?dev=true` to generate certs for your ip/hostname
3. Install the public key of the reverse proxy in the keystore and truststore under the alias `localhost` by running this command in `etc/keystores` `openssl s_client -showcerts -connect localhost:YOUR_PORT_80_PORT_ASSIGNED_BY_DOCKER< /dev/null | openssl x509 -outform DER > derp.der; keytool -delete -alias localhost -keystore serverKeystore.jks -storepass changeit; keytool -import -alias localhost -keystore serverKeystore.jks -file derp.der -storepass changeit -noprompt; keytool -delete -alias localhost -keystore serverTruststore.jks -storepass changeit; keytool -import -alias localhost -keystore serverTruststore.jks -file derp.der -storepass changeit -noprompt`
4. Change IdP for basic auth
5. Shutdown DDF
6. Configure the external properties in `system.properties`
7. Access DDF at `https://localhost:DOCKER_PORT/LOCAL_CONTEXT`. Make sure all the UIs still work normally. Logins, logouts, admin, every iframe, intrigue, standard, simple.

#### What are the relevant tickets?
[DDF-3686](https://codice.atlassian.net/browse/DDF-3686)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by vinamartin** [comment]
I've manually verified the UI works with this change.

**Comment by djblue** [comment]
We might also need to re-wire all the `main.js` require.js configs to use relative paths. As of right now a lot of them use absolute `/webjars` paths.

**Comment by brendan-hofmann** [comment]
@SmithJosh thanks for the detailed feedback!

**Comment by brendan-hofmann** [comment]
build now

File: ui/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+src/main/resources/webapp
+packages/codice-icons/icons
+test.js
+server*.js
+webpack*
+cometd.js
+cometd-web-example
+packages/linter
+diable-idp.js
```
Comment (stustison) [inline ui/.eslintignore]: typo? `diable-idp.js`

Comment (brendan-hofmann) [reply ui/.eslintignore]: typo

File: ui/packages/linter/lib/rules/no-absolute-urls.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        while (node.type === 'BinaryExpression') {
+          node = node.left
+        }
+        let cometd = context.getAncestors().filter(parent => parent.type === 'CallExpression' && parent.callee && parent.callee.object && parent.callee.object.object && parent.callee.object.object.name === 'Cometd')
+        if (cometd.length === 0 && node.type === 'Literal' && typeof node.value === 'string' && node.value.startsWith('/')) {
+          context.report(node, 'Do not concatenate absolute URLs')
+        }
+      },
+      Literal (node) {
+        let cometd = context.getAncestors().filter(parent => parent.type === 'CallExpression' && parent.callee && parent.callee.object && parent.callee.object.object && parent.callee.object.object.name === 'Cometd')
```
Comment (shaundmorris) [inline ui/packages/linter/lib/rules/no-absolute-urls.js]: Hopefully we will remove cometD sooner than later

File: ui/packages/standard/src/main/resources/index.html [hunk 3]
```diff
@@ -25,7 +25,7 @@
     <meta name="author" content="">
 
     <style>
-        @import url(/webjars/cesiumjs/1.22.0/Cesium/Widgets/CesiumWidget/CesiumWidget.css);
+        @import url(../../webjars/cesiumjs/1.22.0/Cesium/Widgets/CesiumWidget/CesiumWidget.css);
```
Comment (shaundmorris) [inline ui/packages/standard/src/main/resources/index.html]: Hopefully standard search UI removed sooner than later.

**Comment by brendan-hofmann** [comment]
Action providers are mostly handled by the changes to `SystemBaseUrl`. @vinamartin 's initial internal/external URL work helped a lot there. The only places that should actually need to know how URLs are constructed would be where we use system properties for blueprint/metatype configuration.

**Comment by brendan-hofmann** [comment]
build now

**Comment by brendan-hofmann** [comment]
Exam setup failure, `Unable to generate system subject`. These passed last time though and the kml endpoint tests passed this time.

**Pull request merged**